subprojects/language-java/src/testFixtures/groovy/org/gradle/language/fixtures/NonIncrementalProcessorFixture.groovy

```diff
@@ -31,8 +31,15 @@ import groovy.transform.CompileStatic
 @CompileStatic
 class NonIncrementalProcessorFixture extends AnnotationProcessorFixture {
 
-    NonIncrementalProcessorFixture() {
+    private final boolean providesNoOriginatingElements
+    private final boolean readsResources
+    private final boolean writesResources
+
+    NonIncrementalProcessorFixture(boolean providesNoOriginatingElements, boolean readsResources, boolean writesResources) {
```

Comment:
It's certainly does not block this PR, but I find the places where this constructor is used a bit hard to read. How about passing a `Map` and using Groovy syntax for "named arguments"? Or adding additional `with()` style methods?

Reply:
Fixed